### PR TITLE
Derek task4 sprint2: Dynamic Automation of IP Address

### LINKED ISSUE #52: [Task 4] Update IP Address Automation with template.json

### PR DESCRIPTION
**Pull request by elb-dev**
# What:
Creates a local IP address for use in the cloud
# How to use:
## Local testing:

Prerequisite: Powershell

- Open a Powershell terminal
- Navigate to automation/ip-address-script
- Run ip-create.ps1
- Optional: Include desired parameters in the run command, example: `./create-ip.ps1 -Tenant #### -Subscription #### -ResourceGroupName resource-group-name -IpName myIpAddress`
- Follow instructions with subscription id, tenant id, username, and password
- Most parameters not required. Those not included in run command which are will be requested. All else will be set from default values.

## Why:
Cloud projects need an IP address, and it needs to be easily deployable

# Time:
1 hour researching
4 hours writing and testing
2 hours testing and reviewing team code

### CODE REVIEW

**Comment by asiagromasia** [comment]
Adding allowed options to IpVersion, server location and method would make the code more safe.

File: automation/ip-address-script/template.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "ipName": {
+            "defaultValue": "nsc-rg-dev-usw2-ip",
+            "type": "String"
+        },
+        "serverLocation": {
+            "defaultValue": "westus2",
```
Comment (asiagromasia) [inline automation/ip-address-script/template.json:10]: You could add here AllowedValues to limit the input possibilities

File: automation/ip-address-script/template.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            "type": "string"
+        },
+        "IpVersion": {
+            "defaultValue": "IPv4",
```
Comment (asiagromasia) [inline automation/ip-address-script/template.json:50]: add allowValues here would be nice as well

File: automation/ip-address-script/template.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            "type": "string"
+        },
+        "IpMethod": {
+            "defaultValue": "Dynamic",
```
Comment (asiagromasia) [inline automation/ip-address-script/template.json:58]: add option here would limit the input possibilities

**Comment by asiagromasia** [comment]
adding output parameter to template.json would help with referencing this resource

File: automation/ip-address-script/template.json (hunk 1 above)
Comment (sunshinejen) [reply automation/ip-address-script/template.json:10]: Good idea!

**Review by sunshinejen — APPROVED** [review]
tested and verified deployment to Azure

**Pull request merged**